### PR TITLE
Update the installation docs to mention the CPU nodes minimum necessary CPU/memory resources

### LINKED ISSUE #355: Update installation docs to mention the minimum CPU/memory resources required for JobSet controller manager deployment
Since we have bumped up the CPU/memory resources required for the controller manager deployment, some cloud providers default node machine type no longer have sufficient resources to run the controller manger.

As a result, users are more commonly installing JobSet and hitting a [webhook not available](https://github.com/kubernetes-sigs/jobset/blob/main/docs/faq/README.md#1-webhook-not-available-error-when-attempting-to-create-a-jobset) error when deploying a JobSet, and getting confused.

These changes add update the docs with a bolded note indicating they need a machine type with 2 CPUs and 512MB memory available for the controller manager Deployment pod, emphasizing the fact that the default node machine type on cloud providers often does not have enough resources to run both the required kube-system pods and the JobSet controller manager.

### PR DESCRIPTION
**Pull request by danielvegamyhre**
Fixes #355

### CODE REVIEW

File: docs/setup/install.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 Make sure the following conditions are met:
 
 - A Kubernetes cluster with version 1.21 or newer is running. Learn how to [install the Kubernetes tools](https://kubernetes.io/docs/tasks/tools/).
+- Your cluster has at least 1 node with 2+ CPUs and 512+ MB of memory available for the JobSet controller manager Deployment to run on. **NOTE**: On some cloud providers, the default node machine type will not have sufficient resources to run the JobSet controller manager and all the required kube-system pods, so you'll need to use a larger
```
Comment (kannon92) [inline docs/setup/install.md:8]: In your issue, you mention that you wanted the note be bolded.

In this case, **Note** is bolded and the rest of the text is normal whitespace.

Comment (danielvegamyhre) [reply docs/setup/install.md]: Updated

**Comment by kannon92** [comment]
I can't remember the tide command to squash these commits on merge. Can you squash them down to 1?

/hold
/lgtm

**Pull request merged**